shm/Map.cpp

```diff
@@ -0,0 +1,59 @@
+#include "Map.hpp"
+
+Map::Map()
+{
+    islands_.reserve(ISLANDS_COUNT);
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_real_distribution<> distrib(COORDINATE_MIN, COORDINATE_MAX);
```

Comment:
I think this should be a member of the class. Creating a random device based on `mt19937` take time, and also can make some problems, because we have a limited resource of pseudo numbers.